spinner.sh

```diff
@@ -1,68 +1,77 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-# Author: Tasos Latsas
-
-# spinner.sh
-#
-# Display an awesome 'spinner' while running your long shell commands
+# St. Benno
 #
-# Do *NOT* call _spinner function directly.
-# Use {start,stop}_spinner wrapper functions
-
-# usage:
+# @project    BashSpinner
+# @author     Tasos Latsas
+# @author     André Lademann <andre.lademann@netresearch.de>
+# @link       http://fitnr.com/showing-a-bash-spinner.html
+# @example
 #   1. source this script in your's
 #   2. start the spinner:
-#       start_spinner [display-message-here]
+#       spinner_start [display-message-here]
 #   3. run your command
 #   4. stop the spinner:
-#       stop_spinner [your command's exit status]
+#       spinner_stop [your command's exit status]
 #
 # Also see: test.sh
-
-
 function _spinner() {
+
     # $1 start/stop
     #
     # on start: $2 display message
     # on stop : $2 process exit status
-    #           $3 spinner function pid (supplied from stop_spinner)
-
-    local on_success="DONE"
-    local on_fail="FAIL"
-    local white="\e[1;37m"
-    local green="\e[1;32m"
-    local red="\e[1;31m"
-    local nc="\e[0m"
+    #           $3 spinner function pid (supplied from spinner_stop)
+    local on_success=" ok "
+    local on_fail="fail"
+    local white="\033[0;37m"
+    local green="\033[01;32m"
+    local red="\033[01;31m"
+    local nc="\033[0m"
 
     case $1 in
         start)
-            # calculate the column where spinner and status msg will be displayed
-            let column=$(tput cols)-${#2}-8
-            # display message and position the cursor in $column column
-            echo -ne ${2}
-            printf "%${column}s"
+            # Calculate the column where spinner and status msg will be displayed
 
-            # start spinner
+            # Start spinner
             i=1
             sp='\|/-'
-            delay=${SPINNER_DELAY:-0.15}
+            delay=${SPINNER_DELAY:-0.25}
+            counter=0
```

Comment:
I get this error with `zsh`
```
$ source fidget.sh

$ spinner_start sleep 1
_spinner:40: unrecognized modifier `i'
```

Reply:
@gianpaj 
Okay, but it's made for bash.